src/core/inventory.c

```diff
@@ -386,4 +403,58 @@ artifact_name_device_type_cb(mender_keystore_t **inventory, uint8_t *inventory_l
     return MENDER_OK;
 }
 
+static mender_err_t
+provides_cb(mender_keystore_t **inventory, uint8_t *inventory_len) {
+    mender_err_t             ret;
+    mender_key_value_list_t *provides = NULL;
+    size_t                   n_provides;
+    uint8_t                  idx;
+    mender_key_value_list_t *item;
+
+    /* mender_storage_get_provides() gives us a linked list (which is a good
+       structure for manipulation with provides data in other places), but we
+       need a key-value array so we need to do a transformation. */
+    if (MENDER_OK != (ret = mender_storage_get_provides(&provides))) {
+        mender_log_error("Failed to get provides from storage");
+        return ret;
+    }
+
+    n_provides = mender_utils_key_value_list_length(provides);
+    if (n_provides > UINT8_MAX) {
+        mender_log_error("Too many provides values, cannot inventory them");
+        ret = MENDER_FAIL;
```

Comment:
Could we truncate it instead?

(related, but I didn't check in the previous to the previous PR: if an inventory callback fails will the rest of the inventory still be submitted?)

Reply:
> Could we truncate it instead?

Sure, I just wasn't sure about the prioritization. Clearly that `rootfs-image....` item is quite important, for example. But we can deal with that later.


> if an inventory callback fails will the rest of the inventory still be submitted?

Yes, it's a best-effort logic.